onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_utils.h

```diff
@@ -210,92 +210,88 @@ class TRTModelMetadefIdGenerator {
       cur_graph = cur_graph->ParentGraph();
     }
 
-    uint32_t instance_hash[4] = {0, 0, 0, 0};
-
     const Graph& main_graph = *cur_graph;
-
-    // hash the bytes in the Graph instance. we can't just use the address as a new Graph instance may use
-    // the same memory (unit tests prove this can occur). the raw bytes of the Graph instance should be a unique
-    // fingerprint for the instance that can use used as the key to the hash of the graph name/inputs&outputs/metadata.
-    MurmurHash3::x86_128(&main_graph, gsl::narrow_cast<int32_t>(sizeof(Graph)), instance_hash[0], &instance_hash);
-    HashValue graph_instance_hash = instance_hash[0] | (uint64_t(instance_hash[1]) << 32);
-
-    // if we've already hashed this main graph instance use the cached value
-    auto entry = trt_main_graph_hash_.find(graph_instance_hash);
-    if (entry != trt_main_graph_hash_.cend()) {
-      model_hash = entry->second;
-    } else {
-      uint32_t hash[4] = {0, 0, 0, 0};
-
-      // Use graph name instead of path to avoid cache regeneration if path changes
-      const auto& model_name_str = main_graph.Name();
-      if (!model_name_str.empty()) {
-        MurmurHash3::x86_128(model_name_str.data(), gsl::narrow_cast<int32_t>(model_name_str.size()), hash[0], &hash);
+    uint32_t hash[4] = {0, 0, 0, 0};
+
+    auto hash_str = [&hash](const std::string& str) {
+      MurmurHash3::x86_128(str.data(), gsl::narrow_cast<int32_t>(str.size()), hash[0], &hash);
+    };
+
+    // Use model name instead of path to avoid cache regeneration if path changes
+    const auto& model_path = main_graph.ModelPath();
+    if (!model_path.IsEmpty()) {
+      // Get model name
+      PathString leaf = model_path.GetComponents().back();
+      std::string model_name = ToUTF8String(leaf.c_str());
```

Comment:
If we hash model name here, will it be a case that a model has different model names and ends up with different hashed ID?

Reply:
yes. If user just changed model name and model itself doesn't change, ORT will treat it as a different model and generate a new engine.